### ISSUE TITLE
Group ID specific information 

### ISSUE THREAD

**Issue by AaronDeRybelHowest** [jdb78/pytorch-forecasting#88]
Hello @jdb78 ,

If possible could u push me in the right direction for following goal, I could write additional code myself if needed.

The values we want to predict (="target") grouped by a unique combination of "group_id's" I call this a "unique time-series".

**My goal is to : Extract predictions, actual values for "target" for each "unique time-series". So I can calculate sMAPE, make plots for each unique time-series.**\
More in detail below

Currently I tested out TFT on the M5 kaggle dataset: https://www.kaggle.com/c/m5-forecasting-accuracy/data
The google colab notebook is here : https://colab.research.google.com/drive/1OSSf7qgOeyZRSUbGgBektHSPSZ1Yambp?usp=sharing

I used 'shop_id' as single group_id creating time-series for **10 unique groups**. I have **19130 samples.** 

### Validation dataset

We can make predictions and show them using the plot function (see code below).\
This shows some samples of the validation dataset but group_id information can't be found.
**My question is: How can val_dataloader be used to get the actual and predicted values of target for each unique time-series ?**

```
# raw predictions are a dictionary from which all kind of information including quantiles can be extracted
raw_predictions, x = best_tft.predict(val_dataloader, mode="raw", return_x=True)

#we can see which information is in these raw predictions
display(raw_predictions.keys())
display(x.keys())

# calculate metric by which to display
predictions = best_tft.predict(val_dataloader)
mean_losses = SMAPE(reduction="none")(predictions, actuals).mean(1)
indices = mean_losses.argsort(descending=False)  # sort losses
for idx in range(10):  # plot 10 examples
    best_tft.plot_prediction(x, raw_predictions, idx=indices[idx], add_loss_to_title=SMAPE())
```

### New data/Test-set

I have kept some data seperate to not include in the training procedure.
I can extract the real values of the target for each unique time-series.
**My Question is: How can I extract predictions on the target for each unique time-series for new data ?**

**Comment by jdb78** [comment]
On the first question: You probably want to use https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.models.base_model.BaseModel.html#pytorch_forecasting.models.base_model.BaseModel.predict with `return_index=True`

The group_id should be used to identify each timeseries directly, i.e. probably you need also a product_id.

For new data, you can use the `from_dataset()` method to create a test set. See https://pytorch-forecasting.readthedocs.io/en/latest/tutorials/stallion.html#Create-dataset-and-dataloaders for an example.

**Comment by AaronDeRybelHowest** [comment]
Hello @jdb78 ,
Thanks for your hints. 
I've sucesfully extracted the forecasts on group id level.

You might find this interesting as additional example. 
I used the TFT model on the M5 kaggle dataset in google colab (notebook below).
[TFT_M5data.ipynb.gz](https://github.com/jdb78/pytorch-forecasting/files/5445378/TFT_M5data.ipynb.gz)

If you think this is interesting for users as example, I could issue a pull request.
Anyways cheers for making this git to access fancy DL forecasting.

**Comment by lee-man** [comment]
Hi @AaronDeRybelHowest,
I go through your m5 preprocessing and training code. It's excellent!

Is there any available experimental result? I am curious about the performance of TFT on M5. Thanks!

**Comment by AaronDeRybelHowest** [comment]
> Hi @AaronDeRybelHowest,
> I go through your m5 preprocessing and training code. It's excellent!
> 
> Is there any available experimental result? I am curious about the performance of TFT on M5. Thanks!

Hi Lee-man, 
I'll include an up-to-date notebook of the results. However currently I use W&B for displaying results it's really neat and free: https://wandb.ai/site

The sMAPE looks pretty good for most shop-id's
![image](https://user-images.githubusercontent.com/43145159/129533718-19b7834a-3b76-4cc2-b74e-41257e7e055d.png)

When making predictions on new data:
![image](https://user-images.githubusercontent.com/43145159/129533931-eee05a6c-20a9-41cb-92ae-dfbb52917e99.png)

The most recent version of the notebook:
[TFT_model.zip](https://github.com/jdb78/pytorch-forecasting/files/6990813/TFT_model.zip)

**Comment by lee-man** [comment]
@AaronDeRybelHowest ,
Thanks for your sharing! I will check your notebook later.

Min

**Comment by LiuNull** [comment]
mark
